### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by bharatkram**
completed the tasks of 17-04

### CODE REVIEW

File: project1/templates/register.html [hunk 1]
```diff
@@ -63,34 +63,55 @@
...
+
+                    <label for="usr"><b>Username</b></label>
+                    <input type="text" placeholder="Enter Username" name="usr" required>
+
+                    <label for="psw"><b>Password</b></label>
+                    <input type="password" placeholder="Enter Password" name="psw" required>
+
+                    <hr>
+                    <input name="but" type = "submit" value = "Login" class="btn">
+                    <input name="but" type = "submit" value = "Register" class="btn">
```
Comment (madhulika9293) [inline project1/templates/register.html]: Both login and register have the same name - "but". Are you able to differentiate? Please test it thoroughly.

File: project1/import.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    f = open("books.csv")
+    reader = csv.reader(f)
+    counter = 0
+    for isbn, title, author, pub_year in reader:
+        if isbn == "isbn":
+            continue
+        counter += 1
+        book = Book(id=counter, isbn=isbn, title=title, author=author, pub_year=int(pub_year))
+        db.session.add(book)
+        print(f"{counter}) Added {title} with {isbn} written by {author} in {pub_year}.")
```
Comment (madhulika9293) [inline project1/import.py:26]: Did you get 5000 print statements for 5000 records inserted? If so, this is unnecessary. Refrain from doing this in future codes.

**Review by madhulika9293 — COMMENTED** [review]
Good job! Make the changes and merge.

**Comment by madhulika9293** [comment]
Please post to 'auth'. Once check the routes they mentioned in LMS.

File: project1/import.py (hunk 2 above)
Comment (bharatkram) [reply project1/import.py:26]: ok

File: project1/README.md [hunk 3]
```diff
@@ -24,4 +24,15 @@ Designed the 'register' page in 3 forms -
...
+# 17-04
+Task 1:
+Added the login button adjacent to the register button on the /register page. Users can register or login from the same page and the same form. Changed the button to input tag from the form tag, and changed the form action from the form tag to this input tag.
+Login button posts the form inputs to the /auth page, whereas register button posts it to the same page.
+
+Task 2:
+Added the session variable on "username", it is None. Everytime a user logs in, it is changed to his name and when no user is logged in. When logged out, it is None again.
+
+Task 3:
+Read the contents of the books.csv file and uploaded in the database. ORM has been used. The class has been defined in the models.py file.
```
Comment (madhulika9293) [inline project1/README.md:38]: Good job with this!

File: project1/application.py [hunk 4]
```diff
@@ -55,6 +45,21 @@ def register():
...
+        return redirect(url_for('register'))
+    name = request.form.get("usr")
+    user = db.execute("SELECT * FROM users WHERE username = :name", {"name": name}).fetchone()
+    print(user)
+    if user is not None:
+        psw = request.form.get("psw")
+        if user[1] == psw:
+            session["username"] = user[0]
+            return redirect(url_for('userhome'))
+    return render_template("register.html", act = 0.1)
```
Comment (madhulika9293) [inline project1/application.py:60]: this works!

**Review by madhulika9293 — COMMENTED** [review]
You can go ahead and merge now!

**Pull request merged**